File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long max_num=(long)Math.sqrt(c);
        long l=0;
        long r=max_num;
        while(l<=r){
           long ans=l*l +r*r;

           if (ans>(long)c){
            r--;
           }
           else if(ans<(long)c){
            l++;
           }
           else{
            return true;
           }
        }
        return false;
        
    }
}

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        max_num=int(sqrt(c))
        l,r=0,max_num

        while l<=r:
            ans=l**2+r**2
            if ans>c:
                r-=1
                
            elif ans<c:
                l+=1
            else:
                return True
        return False


        